src/mgmt/srv_drpc.c

```diff
@@ -697,6 +698,8 @@ void ds_mgmt_drpc_pool_set_prop(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 		drpc_resp->body.data = body;
 	}
 
+	if (req->value_case == MGMT__POOL_SET_PROP_REQ__VALUE_STRVAL)
+		D_FREE(resp.strval);
```

Comment:
IMO better to just free it. The resp is zeroed initially, so if it wasn't set, it will be NULL.

Reply:
Done

Reply:
Hmm. I get a segfault if I don't guard the free. Is it a problem to do it this way?

Reply:
Oh, in the generated code it's a union. OK, yeah, you need to guard it. Sorry for the churn.